src/main/java/entities/Topic.java

```diff
@@ -0,0 +1,71 @@
+package entities;
+
+import net.dv8tion.jda.api.entities.Member;
+
+import java.util.LinkedList;
+
+/**
+ * A topic for a server. Internally contains a queue of Members.
+ */
+public class Topic {
+    private final String name;
+    private final LinkedList<Member> queue = new LinkedList<>();
```

Comment:
In Java, the `final` modifier prevents a variable from being reassigned or assigned after instance construction. It prevents us, for example, from calling `this.queue = new LinkedList<>()` in any instance method, since it would be better to empty the existing list than to create a new one. It's not required, but it follows the idea that developers should be given as few methods as possible to modify an object's state, similar to why we use getter/setter methods instead of making fields public.

Reply:
Gotcha! 
After using a language like Swift the concept of getters/setters stops being useful -- the theoretical of "a developer Might change an object they already have read/write access too" doesn't apply to self-contained projects (i.e., not an API), and at that point a Lot of boilerplate goes away :)

Reply:
I do agree that we don't *need* to use as many restrictions because we aren't developing with the intent of interaction with other people's code. However, restrictions like these also keep us from making mistakes as we move forward (e.g. accidentally changing the name attribute of a Topic class instead of recreating the object with a new name), so I feel that stuff like this does help us avoid issues in the long run.